### PR TITLE
xbps-src: detect two types of build loops and abort()

### PR DESCRIPTION
**Pull request by xtraeme**
- depends on itself
- dep depends on targetpkg

### CODE REVIEW

File: common/xbps-src/shutils/build_dependencies.sh [hunk 1]
```diff
@@ -368,6 +383,11 @@ install_pkg_deps() {
                 echo "   [runtime] ${_realpkg}: not found."
             fi
         fi
+        if [ "${_realpkg}" = "$targetpkg" ]; then
```
Comment (maxice8) [inline common/xbps-src/shutils/build_dependencies.sh]: Won't that break packages like wesnoth that depend on their -data subpackage ?

Comment (xtraeme) [reply common/xbps-src/shutils/build_dependencies.sh]: Let me test

Comment (xtraeme) [reply common/xbps-src/shutils/build_dependencies.sh]: No because self subpkgs are ignored.

Comment (xtraeme) [reply common/xbps-src/shutils/build_dependencies.sh]: Verified, if you've got any other case just let me know.

Tested from scratch xbps build.
Tested wesnoth.
Tested extra-cmake-modules/kcoreaddons with the build loops

All success

**Comment by maxice8** [comment]
On Mon, Apr 15, 2019 at 04:01:24AM -0700, Juan RP wrote:
> xtraeme commented on this pull request.
> 
> 
> 
> > @@ -368,6 +383,11 @@ install_pkg_deps() {
>                  echo "   [runtime] ${_realpkg}: not found."
>              fi
>          fi
> +        if [ "${_realpkg}" = "$targetpkg" ]; then
> 
> No because self subpkgs are ignored.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/10967#discussion_r275308127

Good

**Pull request merged**